### ISSUE TITLE
cant get completion to work with neovim

### ISSUE THREAD

**Issue by lanox** [juliosueiras/vim-terraform-completion#50]
Hi, 

I cant seem to get this working with neovim,

I have installed the following plugins,

```
Plug 'hashivim/vim-terraform'
Plug 'juliosueiras/vim-terraform-completion'
Plug 'w0rp/ale'   < -- although not sure if this is supported as I use this instead on syntastic
```
this is my CheckHealth output

```
health#deoplete#check
========================================================================
## deoplete.nvim
  - OK: exists("v:t_list") was successful
  - OK: has("timers") was successful
  - OK: has("python3") was successful
  - INFO: If you're still having problems, try the following commands:
    $ export NVIM_PYTHON_LOG_FILE=/tmp/log
    $ export NVIM_PYTHON_LOG_LEVEL=DEBUG
    $ nvim
    $ cat /tmp/log_{PID}
    and then create an issue on github

health#nvim#check
========================================================================
## Configuration
  - ERROR: 'paste' is enabled. This option is only for pasting text.
    It should not be set in your config.
    - ADVICE:
      - Remove `set paste` from your init.vim, if applicable.
      - Check `:verbose set paste?` to see if a plugin or script set the option.

## Performance
  - OK: Build type: Release

## Remote Plugins
  - OK: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=^H
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~
  - INFO: $TERM_PROGRAM='iTerm.app'
  - INFO: $COLORTERM='truecolor'

health#provider#check
========================================================================
## Clipboard (optional)
  - OK: Clipboard tool found: pbcopy

## Python 2 provider (optional)
  - WARNING: No Python interpreter was found with the neovim module.  Using the first available for diagnostics.
  - ERROR: Python provider error
    - ADVICE:
      - provider/pythonx: Could not load Python 2:
          /usr/local/bin/python2 does not have the "neovim" module. :help |provider-python|
          /usr/local/bin/python2.7 does not have the "neovim" module. :help |provider-python|
          python2.6 not found in search path or not executable.
          /usr/local/bin/python does not have the "neovim" module. :help |provider-python|
  - INFO: Executable: Not found

## Python 3 provider (optional)
  - INFO: `g:python3_host_prog` is not set.  Searching for python3 in the environment.
  - INFO: Executable: /usr/local/bin/python3
  - INFO: Python3 version: 3.7.0
  - INFO: python3-neovim version: 0.2.6
  - OK: Latest python3-neovim is installed: 0.2.6

## Ruby provider (optional)
  - INFO: Ruby: ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin17]
  - INFO: Host: /usr/local/bin/neovim-ruby-host
  - OK: Latest "neovim" gem is installed: 0.7.1

## Node.js provider (optional)
  - INFO: Node.js: v10.9.0
  - WARNING: Missing "neovim" npm (or yarn) package.
    - ADVICE:
      - Run in shell: npm install -g neovim
      - Run in shell (if you use yarn): yarn global add neovim

```
also have this set in init.vim

```
" (Optional)Remove Info(Preview) window
set completeopt-=preview

" (Optional)Hide Info(Preview) window after completions
autocmd CursorMovedI * if pumvisible() == 0|pclose|endif
autocmd InsertLeave * if pumvisible() == 0|pclose|endif

" (Optional) Enable terraform plan to be include in filter
let g:syntastic_terraform_tffilter_plan = 1

" (Optional) Default: 0, enable(1)/disable(0) plugin's keymapping
let g:terraform_completion_keys = 1

" (Optional) Default: 1, enable(1)/disable(0) terraform module registry completion
let g:terraform_registry_module_completion = 0
let g:deoplete#omni_patterns = {}
let g:deoplete#omni_patterns.terraform = '[^ *\t"{=$]\w*'
let g:deoplete#enable_at_startup = 1
call deoplete#initialize()
```
any ideas on why autocompletion doesn't work ?

thanks

**Comment by esn89** [comment]
I think you might have the same issue as https://github.com/juliosueiras/vim-terraform-completion/issues/49

**Comment by lpuljic** [comment]
@esn89 yeah I saw that but it did not work for me.

**Comment by lanox** [comment]
Well got it further but, i get ``` Omni completion (^O^N^P) Pattern not found``` and neither instructions fro deoplete worked for me.

**Comment by lanox** [comment]
@juliosueiras  any help would be appreciated.

**Comment by lanox** [comment]
Anyone ?

**Comment by bzub** [comment]
Here's a log with some exceptions that seem relevant to this issue, when I try omnicomplete after typing `resource "`

[exceptions.log](https://github.com/juliosueiras/vim-terraform-completion/files/2470960/exceptions.log)

**Comment by juliosueiras** [comment]
updated the repo due to dein.vim load ftplugin very differently from others(it will collect ftplugin and overwrite them)

so pull the newest update of the repo and follow [this](https://asciinema.org/a/AckDTqCDEZyqV8cVyxK8kI5UI)

Neovim version: 0.3.1

**Comment by juliosueiras** [comment]
wait, are you sure?, since the video show completion working

**Comment by lanox** [comment]
Yes, sorry my bad, it shows it’s working, but it is not working for me, I still get the error I mentioned above. 

I have followed readme and did exactly as it states, with out any luck.

**Comment by juliosueiras** [comment]
Just double checking everything

A) what neovim version are you using?

B) what is your plugin manager?

C) you have json gem installed? (or can use `require "json"` in ruby

**Comment by lanox** [comment]
A) what neovim version are you using?
NVIM v0.3.1
B) what is your plugin manager?
Plug
C) you have json gem installed? (or can use require "json" in ruby
gem list shows

json (default: 2.1.0)

**Comment by juliosueiras** [comment]
can you double check this https://asciinema.org/a/CZH4fT2u1IIRUcn3wcFZVWnzk and see I am using the same config as you? (Neovim 0.3.1, same ruby neovim gem version, and json 2.1.0)

**Comment by lanox** [comment]
@juliosueiras I have no idea so I copied your config from asciiname video and created new init.conf.

This is what I have now in my init.conf and its still not working.
Also I am running max os I dunno if that has anything to do with.

```
if &compatible                                                                            
 set nocompatible                                                                         
endif                                                                                     
                                                                                          
" - For Neovim: ~/.local/share/nvim/plugged                                               
" - Avoid using standard Vim directory names like 'plugin'                                
call plug#begin('~/.config/nvim/plugged')                                                  
Plug 'hashivim/vim-terraform'                                                             
Plug 'juliosueiras/vim-terraform-completion'                                              
Plug 'w0rp/ale'                                                                           
Plug 'Shougo/deoplete.nvim'                                                               
call plug#end()                                                                           
                                                                                          
filetype plugin indent on                                                                 
syntax enable                                                                             
                                                                                          
" (Optional)Remove Info(Preview) window                                                   
set completeopt-=preview                                                                  
                                                                                          
" (Optional)Hide Info(Preview) window after completions                                   
autocmd CursorMovedI * if pumvisible() == 0|pclose|endif                                  
autocmd InsertLeave * if pumvisible() == 0|pclose|endif 
                                                                                          
" (Optional) Enable terraform plan to be include in filter                                
let g:syntastic_terraform_tffilter_plan = 1                                               
                                                                                          
" (Optional) Default: 0, enable(1)/disable(0) plugin's keymapping                         
let g:terraform_completion_keys = 1                                                       
                                                                                          
" (Optional) Default: 1, enable(1)/disable(0) terraform module registry completion        
let g:terraform_registry_module_completion = 0     

let g:deoplete#omni_patterns = {}
call deoplete#custom#option('omni_patterns', {
\ 'complete_method': 'omnifunc',
\ 'terraform': '[^ *\t"{=$]\w*',
\})

call deoplete#initialize()
```

**Comment by kinowarrior** [comment]
In the same situation, very similar setup to @lanox

**Comment by kinowarrior** [comment]
syntax & formatting works great, but autocompletion is a no show.

**Comment by kinowarrior** [comment]
With <c-x><c-o> for omnicomplete, I receive `E117: Unknown function: terraformcomplete#Complete
`

**Comment by kinowarrior** [comment]
Well, after quite a bit of digging around, I got everything working great for autocompletion, and ctags / tagbar support too.

![terraform-vim](https://user-images.githubusercontent.com/49199497/57301134-3b04d300-7113-11e9-85fc-7bfd9781bc8e.gif)

In my situation, the problem was related to my path entries being a mess in `.zshrc`. At some point, a `path_helper` entry was added which meant my environment was picking up system/brew version of ruby and not the intended rbenv global env where I had installed neovim gem etc. Sorting out the $PATH issue, and getting the correct ordering resolved this problem.

Environment is: MacOs Mojave, Neovim 0.3.4, with iTerm & Tmux in case it assists others. Very similar plugins setup to the OP.

May try installing neomake to get linting working...  Thanks to all the maintainers for providing this excellent Plugin. Appreciated.
